### ISSUE TITLE
Weird multi-line string indentation

### ISSUE THREAD

**Issue by pfitzseb** [domluna/JuliaFormatter.jl#501]
```
julia> print(JuliaFormatter.format_text("""
       write("", \"\"\"
           second
           third
           \"\"\")
       """))
write(
    "",
    """
second
third
""",
)
```
Not strictly an issue since it parses correctly, but does look weird.

Originally reported at https://github.com/julia-vscode/julia-vscode/issues/2545.

**Comment by domluna** [comment]
this looks correct to me, if you look at how the text is placed in relation to the opening """ in the original text it's the same in the formatted output. It can lead to weird text in some cases, but you can indent the words more as one 1 way to get around it.

This nests because the current rule is that if a string is multiline in a function call, the function call is automatically nested.

**Comment by domluna** [comment]
true, would this backfire in other cases though?

**Comment by pfitzseb** [comment]
Well, Julia disregards
- a leading newline
- the whitespace all lines have in common

Sooo, sure, there are case where indentation isn't possible.

**Comment by domluna** [comment]
would it be bad to align the lines to the first " if they currently appear prior to it? If it's a good idea it wouldn't be difficult to do (i think)

**Comment by mortenpi** [comment]
Not 100% sure if it's related, but this

```julia
function myfunction_foo(io::IO)
    write(io, """
    """)
end
```

goes into

```julia
function myfunction_foo(io::IO)
    write(
        io,
        """
""",
    )
end
```

which does seem incorrect to me.

**Comment by domluna** [comment]
that's intentional. the distance between the start and end quotes stays consistent.

**Comment by mortenpi** [comment]
That doesn't seem necessary? The string wouldn't change? So it could just go into

```julia
function myfunction_foo(io::IO)
    write(
        io,
        """
        """,
    )
end
```

Or if we do insist on keeping the distance, then it could go into

```julia
function myfunction_foo(io::IO)
    write(
        io,
            """
        """,
    )
end
```

which would still be a bit weird, but slightly better? Basically, it should be aligning the ending `"""`, not the starting one.

**Comment by pfitzseb** [comment]
Yeah, that's not required. Julia doesn't care about leading whitespace inside of the string, cf
```
julia> x = """
       """
""

julia> x = """
           """
""

julia> x = """

           """
"\n"

julia> x = """

       """
"\n"

julia> x = """
           ab
           """
"ab\n"

julia> x = """
           ab
       """
"    ab\n"

julia> x = """
        ab
       """
" ab\n"
```

**Comment by domluna** [comment]
```
julia> x = """
           ab
       """
"    ab\n"

julia> x = """
           ab
        """
"   ab\n"

```

these are different based on the positioning of the quotations though, which is why it's maintained.

**Comment by domluna** [comment]
if the other way is more aesthetically pleasing I would be open to it. Not sure when i would get around to doing a PR atm

**Comment by pfitzseb** [comment]
Imho we should never indent multi-line strings by less than the surrounding code. IIRC CSTParser handles multi-line strings with common leading whitespace fairly well, so _maybe_ it'll be easy to implement this here.

**Comment by domluna** [comment]
> Imho we should never indent multi-line strings by less than the surrounding code. IIRC CSTParser handles multi-line strings with common leading whitespace fairly well, so _maybe_ it'll be easy to implement this here.

makes sense. I'd be assist with a PR.
